connectors/src/connectors/notion/index.ts

```diff
@@ -381,17 +381,19 @@ export async function retrieveNotionConnectorPermissions(
     })
   );
 
-  const dbResources: ConnectorResource[] = dbs.map((db) => ({
-    provider: c.type,
-    internalId: db.notionDatabaseId,
-    parentInternalId:
-      !db.parentId || db.parentId === "workspace" ? null : db.parentId,
-    type: "database" as ConnectorResourceType,
-    title: db.title || "",
-    sourceUrl: db.notionUrl || null,
-    expandable: true,
-    permission: "read" as ConnectorPermission,
-  }));
+  const dbResources: ConnectorResource[] = dbs.map((db) => {
+    return {
+      provider: c.type,
+      internalId: db.notionDatabaseId,
+      parentInternalId:
+        !db.parentId || db.parentId === "workspace" ? null : db.parentId,
+      type: "database" as ConnectorResourceType,
```

Comment:
Why do you need this casting here? Seems unnecessary and potentially dangerous. Let me know if I missed something.

Reply:
Yeah it's not great, I believe I do need it inside maps like that. I'm not sure exactly why precisely.

Reply:
Actually it seems I can remove them :+1:

Reply:
I am not sure why you think you need it but I think you don't need it.
When returning from array.map(), you need to explicit the return type somewhere, which you do. So no need to cast here.
You might have needed to cast if you did not have the type explicitly declared before.

Look at my screenshots:
<img width="605" alt="Screen Shot 2023-08-25 at 10 09 45" src="https://github.com/dust-tt/dust/assets/358965/6ea130ff-58e7-45a1-a0eb-bb123c38a089">
<img width="978" alt="Screen Shot 2023-08-25 at 10 10 08" src="https://github.com/dust-tt/dust/assets/358965/4cac5b28-c8c8-448b-bb19-82506f482082">

Reply:
@spolu you need to explicitly say what you are returning at least in one place.
examples:
```
 // here with  ConnectorResource[]
  const pageResources: ConnectorResource[] = await Promise.all( 
// or here in the map return type:
    pages.map((p) : ConnectorResource => {
```